src/codeu/chat/common/InterestSet.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * A collection of all conversations followed by a user, and the users they follow. 
+ *
+ * This class is unfinished and a tad messy as a result, so if you have questions/feedback
+ * that would definitely accelerate things.
+ * 
+ * @author (Sarah Abowitz) 
+ * @version (V1.0.0 | 6.10.17)
+ */
+ 
+  package codeu.chat.common;
+ 
+  import java.io.IOException;
+  import java.io.InputStream;
+  import java.io.OutputStream;
+  import java.util.HashSet;
+ 
+  import codeu.chat.client.core.ConversationContext;
+  import codeu.chat.client.core.MessageContext;
+  import codeu.chat.common.User;
+  import codeu.chat.server.Model;
+  import codeu.chat.util.Serializer;
+  import codeu.chat.util.Serializers;
+ 
+  public final class InterestSet{
+    public static final Serializer<InterestSet> SERIALIZER = new Serializer<InterestSet>() {
+
+      @Override
+      public void write(OutputStream out, InterestSet value) throws IOException {
+
+        Serializers.INTEGER.write(out,value.u);
+        Serializers.INTEGER.write(out,value.b);
+        Serializers.collection(User.SERIALIZER).write(out, value.users);
+        Serializers.collection(Bookmark.SERIALIZER).write(out, value.bookmarks);
+        
+      }
+
+      @Override
+      public InterestSet read(InputStream in) throws IOException {
+
+        InterestSet result = new InterestSet(
+          Serializers.INTEGER.read(in),
+          Serializers.INTEGER.read(in)
+          
+        );
+        
+        result.users.addAll(Serializers.collection(User.SERIALIZER).read(in));
+        result.bookmarks.addAll(Serializers.collection(Bookmark.SERIALIZER).read(in));
+        return result;
+      }
+   }; 
+  
+   public HashSet<User> users; 
+   public HashSet<Bookmark> bookmarks; 
+   
+   // custom sizing variables for users and bookmarks
+   public int u;
```

Comment:
I don't think you need to bother setting these - it looks like a premature optimization to me. I suggest deleting them, and just initializing `users` and `bookmarks` with `new HashSet<>()`, which sets an arbitrary initial size and will be dynamically resized as needed.